### PR TITLE
Fix with OpenCascade 7.8.0.

### PR DESCRIPTION
**Pull request by thierry-FreeBSD**
This should resolves https://github.com/f3d-app/f3d/issues/1304 .

### CODE REVIEW

File: plugins/occt/module/vtkF3DOCCTReader.cxx [hunk 1]
```diff
@@ -29,6 +29,7 @@
 #include <TopoDS_Solid.hxx>
 
 #if F3D_PLUGIN_OCCT_XCAF
+#include <Standard_Version.hxx>
```
Comment (mwestphal) [inline plugins/occt/module/vtkF3DOCCTReader.cxx:32]: you need to move that in alphabetical order

File: plugins/occt/CMakeLists.txt [hunk 2]
```diff
@@ -17,7 +17,13 @@ if("${OpenCASCADE_VERSION}" VERSION_LESS "7.5.0")
   message(FATAL_ERROR "Plugin: OpenCASCADE: ${OpenCASCADE_VERSION} is not supported by F3D, please update your OpenCASCADE installation.")
 endif()
 
-if((NOT TARGET "TKSTEP") AND (NOT TARGET "TKIGES") AND (NOT TARGET "TKMesh"))
+if("${OpenCASCADE_VERSION}" VERSION_LESS "7.8.0")
+  if((NOT TARGET "TKSTEP") AND (NOT TARGET "TKIGES") AND (NOT TARGET "TKMesh"))
```
Comment (mwestphal) [inline plugins/occt/CMakeLists.txt:21]: so these modules disappeared completely in 7.8.0 ? Are they not just renamed ? Should we  check for their presence ?

File: plugins/occt/CMakeLists.txt [hunk 3]
```diff
@@ -17,7 +17,13 @@ if("${OpenCASCADE_VERSION}" VERSION_LESS "7.5.0")
   message(FATAL_ERROR "Plugin: OpenCASCADE: ${OpenCASCADE_VERSION} is not supported by F3D, please update your OpenCASCADE installation.")
 endif()
 
-if((NOT TARGET "TKSTEP") AND (NOT TARGET "TKIGES") AND (NOT TARGET "TKMesh"))
+if("${OpenCASCADE_VERSION}" VERSION_LESS "7.8.0")
```
Comment (mwestphal) [inline plugins/occt/CMakeLists.txt:20]: no need for quotes I believe `${OpenCASCADE_VERSION}`

File: plugins/occt/CMakeLists.txt [hunk 4]
```diff
@@ -17,7 +17,13 @@ if("${OpenCASCADE_VERSION}" VERSION_LESS "7.5.0")
...
+  if((NOT TARGET "TKSTEP") AND (NOT TARGET "TKIGES") AND (NOT TARGET "TKMesh"))
+    message(FATAL_ERROR "Plugin: OpenCASCADE does not contain required modules")
+  endif()
+endif()
+
+if((NOT TARGET "TKDESTEP") AND (NOT TARGET "TKDEIGES") AND (NOT TARGET "TKMesh"))
```
Comment (mwestphal) [inline plugins/occt/CMakeLists.txt:26]: so TKDESTEP is a new module or a module that is shared between all supported versions of OCCT ?

**Comment by mwestphal** [comment]
Ideally we should update the version we use in CI, testing the last release is more important. Let me know if you are ok to do that @thierry-FreeBSD

File: plugins/occt/CMakeLists.txt (hunk 4 above)
Comment (snoyer) [reply plugins/occt/CMakeLists.txt:26]: looks like it `TKSTSEP` and `TKIGES` have been renamed to `TKDESTEP` and `TKDEIGES` according to the initial diff from #1304 
```diff
-if((NOT TARGET "TKSTEP") AND (NOT TARGET "TKIGES") AND (NOT TARGET "TKMesh"))
+if((NOT TARGET "TKDESTEP") AND (NOT TARGET "TKDEIGES") AND (NOT TARGET "TKMesh"))
```

Comment (thierry-FreeBSD) [reply plugins/occt/CMakeLists.txt:26]: > looks like it `TKSTSEP` and `TKIGES` have been renamed to `TKDESTEP` and `TKDEIGES` according to the initial diff from #1304

Yes exactly: the rule is now:

DE TK components have been combined or separated based on specific CAD formats to support plug-in ability.

    Components now have a "TKDE" prefix. The available list includes TKDESTEP, TKDEOBJ, TKDEIGES, TKDEGLTF, TKDEVRML, TKDEPLY, TKDESTL.
    The DE DRAW TK has been updated in a similar way: DRAW components now have a "TKXSDRAW" prefix. The available list includes TKXSDRAWSTEP, TKXSDRAWOBJ, TKXSDRAWIGES, TKXSDRAWGLTF, TKXSDRAWVRML, TKXSDRAWPLY, TKXSDRAWSTL.

**Comment by thierry-FreeBSD** [comment]
ATM most distributions keep OCCT 7.7.x, because they are waiting for upstream of dependent packages to fix them (so do we in FreeBSD, this is why I trying to send these PRs).
A list is available at [Repology](https://repology.org/project/opencascade-occt/versions); it it possible to use one of these rolling releases in you CI?
Or is it possible to modify your CI to build your own release of OpenCascade from sources?

**Comment by Meakk** [comment]
If only the module names have changed, should we just check the OCCT version and build the list of required modules based on the OCCT version?

**Comment by thierry-FreeBSD** [comment]
All the changes are described in the [upgrade guide](https://dev.opencascade.org/doc/overview/html/occt__upgrade.html#upgrade_occt780).

**Comment by mwestphal** [comment]
> ATM most distributions keep OCCT 7.7.x, because they are waiting for upstream of dependent packages to fix them (so do we in FreeBSD, this is why I trying to send these PRs).
> A list is available at [Repology](https://repology.org/project/opencascade-occt/versions); it it possible to use one of these rolling releases in you CI?
> Or is it possible to modify your CI to build your own release of OpenCascade from sources?

Everything is possible. Dont worry about it, Ill take care of it.

File: plugins/occt/CMakeLists.txt (hunk 3 above)
Comment (mwestphal) [reply plugins/occt/CMakeLists.txt:20]: @thierry-FreeBSD please adress this, I will then merge :)

Comment (thierry-FreeBSD) [reply plugins/occt/CMakeLists.txt:20]: > @thierry-FreeBSD please adress this, I will then merge :)

At line #16, there is:
`if("${OpenCASCADE_VERSION}" VERSION_LESS "7.5.0")`
and I did not change it.
Isn't it better to keep the same style?

Comment (mwestphal) [reply plugins/occt/CMakeLists.txt:20]: looks like it was a previous issue. Ok to keep as is.

**Review by mwestphal — APPROVED** [review]

**Comment by mwestphal** [comment]
Thanks for your contribution @thierry-FreeBSD !

**Pull request merged**